packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -42,25 +44,63 @@ const reducer = (state: TickState, action: TickAction): TickState => {
       mode: 'live',
       convertedViewport: convertViewportToMs(state.viewport),
     };
-  } else if (action.type === 'syncViewport') {
-    return stateFromViewport(action.payload);
+  } else if (action.type === 'pause') {
+    return {
+      ...state,
+      mode: 'static',
+    };
+  } else if (action.type === 'updateViewport') {
+    return stateFromViewport(action.viewport);
   }
   return state;
 };
 
 export const useDataZoom = (chartRef: MutableRefObject<EChartsType | null>, viewport: Viewport | undefined) => {
-  const { setViewport, lastUpdatedBy } = useViewport();
+  const { setViewport, group } = useViewport();
   const [{ mode, convertedViewport }, dispatch] = useReducer(reducer, stateFromViewport(viewport));
 
   /**
-   * mode will be used to start the tick interval again
-   * once the user sets the viewport back to relative
-   * we also need to update the viewport incase the user
-   * sets it from the chart prop or viewport picker
+   * function for setting the dataZoom chart option on the echart instance
    */
-  useEffect(() => {
-    dispatch({ type: 'syncViewport', payload: viewport });
-  }, [viewport]);
+  const zoomChart = useCallback(
+    ({ startValue, endValue }: { startValue: number; endValue: number }) => {
+      const chart = chartRef.current;
+
+      chart?.setOption({
+        dataZoom: { ...DEFAULT_DATA_ZOOM, startValue, endValue },
+      });
+    },
+    [chartRef]
+  );
+
+  /**
+   * callback function for the viewport group subscription
+   */
+  const handleViewportUpdate = useCallback(
+    (updatedViewport: Viewport, topic?: string) => {
+      /**
+       * we do not need to update the dataZoom if the event originated from
+       * echarts since that is handled internally by echarts
+       */
+      if (topic === ECHARTS_GESTURE) return;
+
+      const { initial, end } = convertViewportToMs(updatedViewport);
+
+      /**
+       * update the current viewport in the reducer
+       * this ensures that the mode is up-to-date
+       * in the event that the user changes from static -> live
+       * this will restart the tick interval
+       *
+       * this also ensures that the duration is up-to-date
+       * in the event that the user changes from for eg. 1min -> 10min
+       */
```

Comment:
Nit: The formatting for your comments is a little hard to understand. I would try to match a max column length of 80 or 120 to keep width consistent. Also, `/** */` is primarily for doc comments. `/* */` is probably the right format to use when you aren't intending to use TSDoc.